### ISSUE TITLE
fix: no-construct-in-interface error condition

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#198]
### Describe the bug

This rule currently throws errors under the following conditions, which may not be appropriate in some cases. Therefore, the error conditions need to be fixed.

- When specifying a class that inherits from `Construct` or `Stack` as a property of an interface

### Fixed Conditions

- Inherits from Construct
- The class itself implements an interface (e.g., `IBucket`), or the parent class implements an interface
  - The interface matches the class name
    - Matching conditions:
      - Implements an interface with `I` prefix added to the class name (e.g., `Bucket` -> `IBucket`)
      - Implements an interface with `I` prefix added to the class name after removing the `Base` suffix (e.g., `BaseService` -> `IService`)
    - Matching exceptions:
      - Class name is `Resource` or `Construct`

### Examples After Fix

Cases that should trigger errors:

```ts
// export declare class FargateService extends BaseService implements IFargateService
import { FargateService } from "aws-cdk-lib/aws-ecs";

// export declare abstract class BucketBase extends Resource implements IBucket
import { BucketBase } from "aws-cdk-lib/aws-s3";

// export declare class EmailIdentity extends EmailIdentityBase
// declare abstract class EmailIdentityBase extends Resource implements IEmailIdentity
import { EmailIdentity } from "aws-cdk-lib/aws-ses";

interface MyConstructProps {
  readonly service: FargateService; // Error target: FargateService implements IFargateService
  readonly bucket: BucketBase; // Error target: BucketBase implements IBucket (matches after removing Base suffix)
  readonly emailIdentity: EmailIdentity; // Error target: EmailIdentity indirectly implements IEmailIdentity
}
```

Cases that should NOT trigger errors:

```ts
// export declare abstract class BaseLoadBalancer extends Resource
import { BaseLoadBalancer } from "aws-cdk-lib/aws-elasticloadbalancingv2";

// export declare class AwsCustomResource extends Construct implements iam.IGrantable
import { AwsCustomResource } from "aws-cdk-lib/custom-resources";

interface MyConstructProps {
  readonly loadBalancer: BaseLoadBalancer; // Not an error target: does not implement any interface
  readonly customResource: AwsCustomResource; // Not an error target: implemented interface name does not match Construct name
}
```

### eslint-cdk-plugin Version

3.3.0

### Node.js Version

22.18.0

### Language

TypeScript

### Checklist

- [ ] I have searched for similar issues before creating this one
- [ ] I have provided all the necessary information to reproduce the issue
- [ ] I have verified this issue is related to eslint-cdk-plugin and not another dependency
